BUG: Test failed in macOS Monterey due to a bug in default multiprocessing module

### Describe the issue:

In a full test,(numpy.test('full')), using pthread model(the default) with Darwin kernel:

lib/tests/test_io.py::TestSaveTxt::test_large_zip test is FAILED due to an AttributeError.
```
FAILED  - AttributeError: Can't pickle local object 'TestSaveTxt.test_large_zip.<locals>.check_large_zip'
```

The failed function is:
```
    def dump(obj, file, protocol=None):
        '''Replacement for pickle.dump() using ForkingPickler.'''
>       ForkingPickler(file, protocol).dump(obj)
E       AttributeError: Can't pickle local object 'TestSaveTxt.test_large_zip.<locals>.check_large_zip'

file       = <_io.BytesIO object at 0x16fc39ae0>
obj        = <Process name='Process-1' parent=1569 initial>
protocol   = None

/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/reduction.py:60: AttributeError
```

The stack trace at multiprocessing is
```
/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py:121: in start
    self._popen = self._Popen(self)
        self       = <Process name='Process-1' parent=1569 initial>
/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/context.py:224: in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
        process_obj = <Process name='Process-1' parent=1569 initial>
/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/context.py:284: in _Popen
    return Popen(process_obj)
        Popen      = <class 'multiprocessing.popen_spawn_posix.Popen'>
        process_obj = <Process name='Process-1' parent=1569 initial>
/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/popen_spawn_posix.py:32: in __init__
    super().__init__(process_obj)
        __class__  = <class 'multiprocessing.popen_spawn_posix.Popen'>
        process_obj = <Process name='Process-1' parent=1569 initial>
        self       = <multiprocessing.popen_spawn_posix.Popen object at 0x16fbbb070>
/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/popen_fork.py:19: in __init__
    self._launch(process_obj)
        process_obj = <Process name='Process-1' parent=1569 initial>
        self       = <multiprocessing.popen_spawn_posix.Popen object at 0x16fbbb070>
/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/popen_spawn_posix.py:47: in _launch
    reduction.dump(process_obj, fp)
        fp         = <_io.BytesIO object at 0x16fc39ae0>
        prep_data  = {'authkey': b'\xe7\xf6K\x8ceo\xe6se\x82\x11#q\xa9\xde\x19SX\x917\xf0M\x07\xa9-\xfd\x15\xa8~\xc5\x98\x14', 'dir': '/Use...s', 'init_main_from_path': '/Us
ers/vxst/vxst_env_verified/py39_sci_test/tests/0_numpy.py', 'log_to_stderr': False, ...}
        process_obj = <Process name='Process-1' parent=1569 initial>
        resource_tracker = <module 'multiprocessing.resource_tracker' from '/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/m
ultiprocessing/resource_tracker.py'>
        self       = <multiprocessing.popen_spawn_posix.Popen object at 0x16fbbb070>
        tracker_fd = 19
```

Test hardwares include M1 Max with 32 core GPU and 64GB memory and M1 with 16GB memory.

Test python version include cpython 3.8 and cpython 3.9.

Four configurations all fail in the same manner.

### Reproduce the code example:

```python
py.test numpy/lib/tests/test_io.py
```


### Error message:

```shell
../../../Library/Python/3.9/lib/python/site-packages/numpy/lib/tests/test_io.py:598:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py:121: in start
    self._popen = self._Popen(self)
/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/context.py:224: in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/context.py:284: in _Popen
    return Popen(process_obj)
/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/popen_spawn_posix.py:32: in __init__
    super().__init__(process_obj)
/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/popen_fork.py:19: in __init__
    self._launch(process_obj)
/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/popen_spawn_posix.py:47: in _launch
    reduction.dump(process_obj, fp)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

obj = <Process name='Process-1' parent=38364 initial>, file = <_io.BytesIO object at 0x1055a08b0>, protocol = None

    def dump(obj, file, protocol=None):
        '''Replacement for pickle.dump() using ForkingPickler.'''
>       ForkingPickler(file, protocol).dump(obj)
E       AttributeError: Can't pickle local object 'TestSaveTxt.test_large_zip.<locals>.check_large_zip'

/opt/homebrew/Cellar/python@3.9/3.9.13_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/reduction.py:60: AttributeError
```


### NumPy/Python version information:

1.21.6 at both Python 3.8 and Python 3.9 environments, built by homebrew.

```
1.21.6 3.9.13 (main, Aug 28 2022, 18:56:22)
[Clang 13.1.6 (clang-1316.0.21.2.5)]
```
and
```
1.21.6 3.8.13 (default, Aug 28 2022, 18:53:35)
[Clang 13.1.6 (clang-1316.0.21.2.5)]
```

The system uname is:
```
Darwin Mac 21.6.0 Darwin Kernel Version 21.6.0: Wed Aug 10 14:28:23 PDT 2022; root:xnu-8020.141.5~2/RELEASE_ARM64_T6000 arm64 arm Darwin
```

### Context for the issue:

It cause numpy.test('full') to fail on macOS.

--- a/numpy/lib/tests/test_io.py
+++ b/numpy/lib/tests/test_io.py
@@ -13,7 +13,7 @@ from tempfile import NamedTemporaryFile
 from io import BytesIO, StringIO
 from datetime import datetime
 import locale
-from multiprocessing import Process, Value
+from multiprocessing import Value, get_context
 from ctypes import c_bool
 
 import numpy as np
@@ -595,7 +595,12 @@ class TestSaveTxt:
         # Use an object in shared memory to re-raise the MemoryError exception
         # in our process if needed, see gh-16889
         memoryerror_raised = Value(c_bool)
-        p = Process(target=check_large_zip, args=(memoryerror_raised,))
+
+        # Use the same start method between Linux and macOS, the default method of
+        # Linux is 'fork', and the default method of macOS is spawn. 'spawn' will
+        # cause problem in the check_large_zip since the data sharing model is
+        # built on 'fork'
+        p = get_context('fork').Process(target=check_large_zip, args=(memoryerror_raised,))
         p.start()
         p.join()
         if memoryerror_raised.value:

--- a/numpy/lib/tests/test_io.py
+++ b/numpy/lib/tests/test_io.py
@@ -596,11 +596,11 @@ class TestSaveTxt:
         # in our process if needed, see gh-16889
         memoryerror_raised = Value(c_bool)
 
-        # Use the same start method between Linux and macOS, the default method of
-        # Linux is 'fork', and the default method of macOS is spawn. 'spawn' will
-        # cause problem in the check_large_zip since the data sharing model is
-        # built on 'fork'
-        p = get_context('fork').Process(target=check_large_zip, args=(memoryerror_raised,))
+        # Since Python 3.8, the default start method for multiprocessing has 
+        # been changed from 'fork' to 'spawn' on macOS, causing inconsistency 
+        # on memory sharing model, lead to failed test for check_large_zip
+        ctx = get_context('fork')
+        p = ctx.Process(target=check_large_zip, args=(memoryerror_raised,))
         p.start()
         p.join()
         if memoryerror_raised.value:

